### PR TITLE
Migrate formatting_html.py into xarray core

### PR DESCRIPTION
**Pull request by eni-awowale**
Draft:

This PR migrates the `formatting_html.py` module to `xarray/core/formatting_html.py` as part of the on-going effort to merge `xarray-datatree` into `xarray`.

One thing of note is that importing and setting the `OPTIONS` to `"default"` in `datatree/formatting_html.py` ([lines](https://github.com/xarray-contrib/datatree/blob/main/datatree/formatting_html.py#L13%23L15)) were moved into `xarray/core/options.py` on [#L23](https://github.com/pydata/xarray/blob/main/xarray/core/options.py#L23) and [#L49](https://github.com/pydata/xarray/blob/main/xarray/core/options.py#L49). So, I did not add them back to `xarray/core/formatting_html.py`.

- [x]  Completes migration step for `datatree/formating_htmls.py` https://github.com/pydata/xarray/issues/8572
- [x]  Tests added
- [ ]  ~~User visible changes (including notable bug fixes) are documented in whats-new.rst~~
- [ ]  ~~New functions/methods are listed in api.rst~~

### CODE REVIEW

File: xarray/datatree_/datatree/tests/test_formatting_html.py
Comment (eni-awowale) [inline xarray/datatree_/datatree/tests/test_formatting_html.py:1]: Updated to delete `formatting_html.py` and `test_formatting_html.py` in `datatree_`.

File: xarray/core/datatree.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
 from xarray.core.coordinates import DatasetCoordinates
 from xarray.core.dataarray import DataArray
 from xarray.core.dataset import Dataset, DataVariables
+from xarray.core.formatting_html import (
```
Comment (eni-awowale) [inline xarray/core/datatree.py:21]: Updated import statements

File: xarray/core/formatting_html.py [hunk 2]
```diff
@@ -341,3 +343,128 @@ def dataset_repr(ds) -> str:
     ]
 
     return _obj_repr(ds, header_components, sections)
+
+
+def summarize_children(children: Mapping[str, Any]) -> str:
```
Comment (flamingbear) [inline xarray/core/formatting_html.py:348]: I might find a way to identify these moved files as datatree specific. 
e.g `summarize_datatree_children` and `datatree_node_repr`, `_wrap_datatree_repr`.   pick what makes sense to you and use that through out.

**Review by flamingbear — COMMENTED** [review]
I don't have strong opinions, this looks reasonable to me.  If this were my PR I'd change some of the names since the new functions are merged into existing files and I'd want to make sure people knew the fxns were for datatree

File: xarray/tests/test_formatting_html.py [hunk 3]
```diff
@@ -196,3 +197,200 @@ def test_nonstr_variable_repr_html() -> None:
...
     assert "<dt><span>22 :</span></dt><dd>bar</dd>" in html
     assert "<li><span>10</span>: 3</li></ul>" in html
+
+
+@pytest.fixture(scope="module", params=["some html", "some other html"])
+def repr(request):
+    return request.param
+
+
+class Test_summarize_datatree_children:
```
Comment (eni-awowale) [inline xarray/tests/test_formatting_html.py:207]: I renamed these test classes so they would be more descriptive.

**Review by TomNicholas — COMMENTED** [review]
Thankyou for working on this @eni-awowale ! The main question I have before diving deeper is whether or not you have any time to look into the annoying bug which squeezes the width of the html output - see https://github.com/xarray-contrib/datatree/issues/91

**Comment by flamingbear** [comment]
This was merged to the main repo https://github.com/pydata/xarray/pull/8930

**Pull request closed without merging**